src/snowflake/snowpark/files.py

```diff
@@ -50,19 +53,24 @@ def open(
         file_location: str,
         mode: str = "r",
         is_owner_file: bool = False,
+        *,
+        require_scoped_url: bool = True,
     ) -> SnowflakeFile:
         """
         Returns a :class:`~snowflake.snowpark.file.SnowflakeFile`.
         In UDF and Stored Procedures, the object works like a Python IOBase object and as a wrapper for an IO stream of remote files. The IO Stream is to support the file operations defined in this class.
 
-	All files are accessed in the context of the UDF owner and therefore when accessing call files user should prefer setting is_owner_file to default false value as it only allows scoped URLs and therefore a caller cannot inadvertently or maliciously access the owner's files.
+	All files are accessed in the context of the UDF owner and therefore when accessing caller files the user should prefer setting require_scoped_url to True (default value) as it requires the use of a scoped URL and therefore a caller cannot inadvertently or maliciously access the owner's files.
+
+        is_owner_file will be deprecated, please use require_scoped_url instead.
 
         Args:
             file_location: A string of file location. It can be a remote URL/URI.
             mode: A string used to mark the type of an IO stream.
-            is_owner_file: A boolean value, if True, the API is intended to access owner's files and all url/uri are allowed. If False, the API is intended to access files passed into the function by the caller and only scope url is allowed.
+            is_owner_file: A boolean value, if True, the API is intended to access owner's files and all url/uri are allowed. If False, the API is intended to access files passed into the function by the caller and only scoped url is allowed.
+            require_scoped_url: A boolean value, if False, the API is intended to access owner's files and all url/uri are allowed. If True, the API is intended to access files passed into the function by the caller and only scoped url is allowed.
```

Comment:
Let's be explicit here. 
A boolean value, if True requires that the file_location is a scoped URL. This ensures that this URL cannot access UDF owner files that the caller does not have access too.